src/app/encoder.cpp

```diff
@@ -162,6 +162,20 @@ uint16_t encodeReadAttributesCommand(uint8_t * buffer, uint16_t buf_length, uint
     }                                                                                                                              \
     return result;
 
+#define COMMAND_INSERT_STRING(name, str)                                                                                           \
+    size_t length = strlen(str);                                                                                                   \
+    if (!CanCastTo<uint8_t>(length))                                                                                               \
+    {                                                                                                                              \
+        ChipLogError(Zcl, "Error encoding %s command. String too long: %d", name, length);                                         \
+        return 0;                                                                                                                  \
+    }                                                                                                                              \
+    uint8_t str_length = static_cast<uint8_t>(length);                                                                             \
+    buf.Put(str_length);                                                                                                           \
+    for (uint8_t i = 0; i < str_length; i++)                                                                                       \
+    {                                                                                                                              \
+        buf.Put(static_cast<uint8_t>(str[i]));                                                                                     \
```

Comment:
Is there a reason not to call the `const char *` overload of `Put` here?  Seems like it does the same thing....

Reply:
Thanks ! I missed it !